samples/drivers/flash_shell/src/main.c

```diff
@@ -233,23 +232,19 @@ static int cmd_flash(const struct shell *shell, size_t argc, char **argv)
 	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	shell_help_print(shell, NULL, 0);
+	shell_help_print(shell);
```

Comment:
isn't this function redundant now? It's for root command only to print help.

Reply:
Currently there are following ways to print help:

1. command handler can simply call `shell_help_print` (will be renamed to shell_help).
2. User will call command without a handler. Deepest found help message will be displayed.
3. User will call command with a wrong argument count and set Kconfig option: `CONFIG_SHELL_HELP_ON_WRONG_ARGUMENT_COUNT`

I was thinking that user might need to print help in some cases hence I decided to keep it in the API.